.github/workflows/pir_end_to_end_tests.yml

```diff
@@ -0,0 +1,162 @@
+name: PIR E2E Tests
+
+on:
+  workflow_dispatch:
+  schedule:
+     - cron: '0 3 * * 1-5' # 3AM UTC offsetted to legacy to avoid action-junit-report@v4 bug
+  pull_request:
+      
+jobs:
+  pir-e2e-tests:
+    name: PIR e2e tests
+    runs-on: ${{ matrix.runner }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runner: [macos-14-xlarge]
+        include:
+          - xcode-version: "15.4"
+            runner: macos-14-xlarge
+
+    if: |
+      startsWith(github.event.pull_request.base.ref, 'release/') ||
+      startsWith(github.event.pull_request.base.ref, 'hotfix/') ||
+      (contains(github.event.pull_request.body, '- [x] Run PIR E2E tests') && (github.event.action != 'edited' || !contains(github.event.changes.body.from, '- [x] Run PIR E2E tests'))) ||
+      github.event_name == 'schedule'
+
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}-${{ matrix.runner }}
+      cancel-in-progress: true
+
+    timeout-minutes: 20
+
+    steps:
+    - name: Register SSH keys for certificates repository and PIR fake broker repository access
+      uses: webfactory/ssh-agent@v0.7.0
+      with:
+        ssh-private-key: |
+                        ${{ secrets.SSH_PRIVATE_KEY_FASTLANE_MATCH }}
+                        ${{ secrets.SSH_PRIVATE_KEY_PIR_FAKE_BROKER }}
+
+    - name: Check out the PIR fake broker code
+      uses: actions/checkout@v4
+      with:
+        repository: DuckDuckGo/pir-fake-broker
+        ssh-key: ${{ secrets.SSH_PRIVATE_KEY_PIR_FAKE_BROKER }}
+        ref: loremattei/ci-integration
```

Comment:
😅 This was a mostly a test... and I see it's now an outdated branch, so we should probably merge or rewrite and point to main. 
I'll follow up in the fake broker repo to get this sorted out.

Reply:
Ah thanks for flagging, I completely forgot about this